tests/integration/test_prometheus_integration.py

```diff
@@ -39,14 +41,31 @@ async def labels(self) -> List[str]:
         return result["data"] if result["status"] == "success" else []
 
 
+@pytest.fixture
+def updated_charmlib():
+    """Provide (and clean) the Pushgateway's charmlib for the testing charm."""
+    testingcharm_path = Path("tests") / "testingcharm"
+    dest_charmlib = testingcharm_path / CHARMLIB_PATH
+    dest_charmlib.parent.mkdir(parents=True)
+    try:
+        dest_charmlib.hardlink_to(CHARMLIB_PATH)
+        yield
+    finally:
+        shutil.rmtree(dest_charmlib.parent)
+
+
 @pytest.mark.abort_on_fail
-async def test_prometheus_integration(ops_test: OpsTest):
+async def test_prometheus_integration(ops_test: OpsTest, updated_charmlib: None):
     """Validate the integration between the Pushgateway and Prometheous."""
     prometheus_app_name = "prometheus"
     tester_name = "testingcharm"
     apps = [APP_NAME, prometheus_app_name, tester_name]
+
+    # build the pushgateway charm (main service)
     pushgateway_charm = await ops_test.build_charm(".")
```

Comment:
We should probably start using the [`@timed_memoizer`](https://github.com/canonical/grafana-k8s-operator/blob/0a4430d2ddbe448a7933d64b645174889ba0b6e0/tests/integration/conftest.py#L74)

Reply:
Yes! Done, thanks.